memstore/common/upsert_batch.go

```diff
@@ -67,6 +67,21 @@ func (c *columnReader) ReadValue(row int) (unsafe.Pointer, bool) {
 	return unsafe.Pointer(&c.valueVector[row*DataTypeBits(c.dataType)/8]), true
 }
 
+// ReadArrayValue returns the ArrayValue from upsert batch at given row
+func (c *columnReader) ReadArrayValue(row int) (unsafe.Pointer, bool) {
+	validity := c.readValidity(row)
```

Comment:
is this validity of the array itself? I thought we don't have invalid arrays

Reply:
yeah, array should have validity too